src/transformation/visitors/class/index.ts

```diff
@@ -282,29 +285,61 @@ function transformClassLikeDeclaration(
         }
     }
 
+    const noPrototype = isExtension || isMetaExtension;
+
+    // Decorate accessors
```

Comment:
I think it would be better to refactor it into a single-pass loop over class members. Currently decorator evaluation order would be different from TypeScript ([Playground](https://www.typescriptlang.org/play/index.html?experimentalDecorators=true&target=1#code/MYGwhgzhAECC0G8BQ1XQAIApMEpoF4A+aYAewDsJSQBTAOhFIHNMByAWxoBcALUgEwCMrHDhRpOvAYNyJoAXyTjUTbtFVcuNAE4y8CBcuhGssoiQpVaDZmw1btIsWmgQuYLgEtg6tfZ0ATLIGikYm2HjmZJTU9IwsrAAO2qSJTkaJAK4ARiDe0MmpYS6mkcTRVnG2HNx8-AHpLpJ1QfqK8kA)). It also should make this code easier to read.

```ts
const decorationStatements = [];
for (const member of classDeclaration.members) {
	if (ts.isAccessor(member)) {
		result.push(transformAccessor(member));
		if (member.decorators) decorationStatements.push(...)
	} else if (ts.isMethodDeclaration(member)) {
		result.push(transformMethod(member));
	} else ...
}
result.push(...decorationStatements);
```